### ISSUE TITLE
Produce more and more small blob file  while running integrated blobdb.

### ISSUE THREAD

**Issue by GOGOYAO** [facebook/rocksdb#10657]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://groups.google.com/forum/#!forum/rocksdb or https://www.facebook.com/groups/rocksdb.dev

### Expected behavior
blob file size is about the same as the setting(`Options.blob_file_size: 536870912`).
### Actual behavior
many blob files is in small size(less than 100k). 
### Steps to reproduce the behavior
1. setting is as follows. 
```
Options.write_buffer_size: 16777216
Options.blob_file_size: 536870912
```
2. run the integrated blobdb.

**Comment by ltamasi** [comment]
This is expected; the `blob_file_size` option is an upper limit on the size of blob files, i.e. BlobDB starts a new file when this size is reached. So it doesn't mean that all blob files will be of the configured size. Note that BlobDB writes blob files during flushes and compactions; with your configuration where memtables are smaller than `blob_file_size`, flushes in particular have no chance of creating blob files this large. When it comes to blob files written by compactions, they contain the blobs relocated during GC by that specific compaction, and how many live GC-eligible blobs any given compaction encounters is a function of your workload and configuration.

**Comment by ltamasi** [comment]
I also see that you're using a large number of subcompactions; note that each of those will write its own set of SST files and blob files, which is why the compaction wrote 6 very small files instead of 1 slightly larger one.

**Comment by GOGOYAO** [comment]
When compactions happened, I think the blob files selected in this compaction job, will be merged into one blob file. So if there are few data are garbage, along with the compactions going on, the blob file will reach the size I set.
Am I right?

**Comment by ltamasi** [comment]
No, that's not the way it works. GC is integrated with the LSM tree compaction; it is still SST files that get picked for compaction (not blob files), and whatever GC-eligible live blobs those SSTs refer to get opportunitistically relocated.

**Comment by GOGOYAO** [comment]
> No, that's not the way it works. GC is integrated with the LSM tree compaction; it is still SST files that get picked for compaction (not blob files), and whatever GC-eligible live blobs those SSTs refer to get opportunitistically relocated.

So, `whatever GC-eligible live blobs those SSTs refer to`, will relocated into one blob file, which is equivalent to merge some blob files(selected by compaction job and is gc eligible) into one blob file.

**Comment by ltamasi** [comment]
> So, `whatever GC-eligible live blobs those SSTs refer to`, will relocated into one blob file, which is equivalent to merge some blob files(selected by compaction job and is gc eligible) into one blob file.

No. By "live blobs" I mean individual blob *values*, not blob *files*. Only the individual GC-eligible blob values referenced by compacted SSTs are relocated. Any other blob values in those blob files are unaffected.

**Comment by GOGOYAO** [comment]
So any advice or plan to prevent too many blob files? @ltamasi
